routes/courseRoutes.js

```diff
@@ -437,4 +453,63 @@ router.get("/course/delete_all", requireLogin, async (req, res) => {
   res.send("Completed");
 });
 
+// User route
+router.post("/user/", async (req, res) => {
+  const { googleID } = req.body;
+
+  const user = new UserModel({
+    googleID: googleID,
+    email: email,
+    password: password,
+    joinedAt: Date.now(),
+    modifiedAt: Date.now(),
+    subscriptions: []
+  });
+
+  try {
+    await user.save();
+    res.send(user);
+  } catch (err) {
+    res.status(422).send(err);
+  }
+});
+
+// Subscribe to course TODO: check for duplicates
+
+router.post("/course/:id/subscribe",  async (req, res) => {
+  const { user_id, course_id} = req.body;
+
+  (await User.findOneAndUpdate(
+    { _id: user_id }, 
+    { $push: { subscriptions: course_id} }))
+    .save;
+
+  user = await User.findById(user_id);
+  res.send(user)
+
+});
+
+router.post("/user/subscriptions/unsubscribe",  async (req, res) => {
+  const { user_id, course_id} = req.body;
+
+  (await User.findOneAndUpdate(
+    { _id: user_id }, 
+    { $pull: { subscriptions: course_id} }))
+    .save;
+
+  user = await User.findById(user_id);
+  res.send(user)
+
+});
+
+// Get users subscriptions
+router.get("/user/subscriptions/getAll", async (req, res) => {
```

Comment:
This URI doesn't match the standard we are following:
- The request type is spelled out
- It contains capitalization (separate with dashes instead)
- It uses plural instead of singular

Proposed change: `/user/subscription/all`